src/context/DialogManagerContext.tsx

```diff
@@ -12,11 +37,28 @@ const DialogManagerProviderContext = React.createContext<
   DialogManagerProviderContextValue | undefined
 >(undefined);
 
+/**
+ * Marks the portal location
+ * @param children
+ * @param id
+ * @constructor
+ */
 export const DialogManagerProvider = ({
   children,
   id,
 }: PropsWithChildren<{ id?: string }>) => {
-  const [dialogManager] = useState<DialogManager>(() => new DialogManager({ id }));
+  const dialogManager = useMemo<DialogManager>(
+    () => (id && getDialogManager(id)) || new DialogManager({ id }),
+    [id],
+  );
+
+  addDialogManager(dialogManager);
```

Comment:
Should this go inside the `useEffect` instead?

Reply:
It still feels like an anti-pattern. Can we structure this differently so we keep the DialogManagerProvider render pure?

Reply:
@oliverlaz I have converted the DialogManagerProvider into a pure component: https://github.com/GetStream/stream-chat-react/pull/2792/commits/db7b0a7a7d804c900bceab66ee4acee4579eca13